astropy/coordinates/representation.py

```diff
@@ -223,6 +223,26 @@ def _unitstr(self):
                            for component in self.components]))
         return unitstr
 
+    def __eq__(self, other):
+        """
+        Check for equality with another representation.
```

Comment:
The point about the equality being the same shape as the representation should be mentioned here (although also see my comment inline).

Actually, this point should maybe even be added to the narrative docs, because it's behavior that is not really like anything else (it's sensible, but by no means obvious).